Tunnel.js

```diff
@@ -276,22 +312,7 @@ Tunnel.prototype = util.mixin(Object.create(_super), /** @lends module:digdug/Tu
 			}
 
 			var request = sendRequest(self.url, { proxy: self.proxy });
-			request.then(
-				function (response) {
-					var decompressor = new Decompress();
-					decompressor.src(response.data)
-						.use(Decompress.zip())
-						.use(Decompress.targz())
-						.dest(self.directory)
-						.run(function (error) {
-							if (error) {
-								reject(error);
-							}
-							else {
-								resolve();
-							}
-						});
-				},
+			request.then((self.dontExtract ? writeFile : decompressFile),
```

Comment:
I hadn't intended for the user to use `options`; it was just a way to get around the weird `this` binding when a subclass called `download` in the super class. A subclass would pass in options to override the download URL, which would allow `download` in SeleniumTunnel` to call `Tunnel#download` with the four different files it needs to.

If we're going to pull most of the logic out of `download` (which seems reasonable enough), then there should no longer be a need for the `options` parameter on `download`. `Tunnel#download` can keep its existing API and call `_downloadFile` internally. We could probably drop `_postDownload`, too; subclasses would just need something like `_downloadFile(options).then(function () { /* write or expand */ })` (well, 4 of those for SeleniumTunnel). At least in SeleniumTunnel's case that'll avoid the acrobatics that would be required to get a `_postDownload`-style system working with multiple downloads.